Makefile

```diff
@@ -118,6 +118,16 @@ test: ## Run tests
 test-integration: ## Run integration tests
 	source ./scripts/fetch_ext_bins.sh; fetch_tools; setup_envs; go test -v -tags=integration ./test/integration/...
 
+.PHONY: test-e2e-images
```

Comment:
I think we should split building images from running e2e, so someone can run e2e test without building images.
What about having:

- `make docker-build-e2e` (that makes all the images required for e2e tests)
- `make test-e2e` (that only runs test)

Reply:
@fabriziopandini Can we also incorporate the image build improvements from the other PR here?

Reply:
Is this the docker build PR https://github.com/kubernetes-sigs/cluster-api/pull/2965/files?

Reply:
@vincepri 
IMO we should keep the two PRs separated.

In 2965 I still don't have a good solution for:
- avoiding the docker file proliferation
- preserving the docker-build target, the docker-build-[component] targets without introducing a new build target
- deleting cluster-api/builder:dev after everything is completed